chapter/1/rpc-reviewed-dont-keep.md

```diff
@@ -0,0 +1,619 @@
+implementations.
+
+#### XML-RPC and SOAP
+
+SOAP (Simple Object Access Protocol) is a successor of XML-RPC as a web-services
+protocol for communicating between a client and server. It was initially
+designed by a group at Microsoft {% cite soaparticle1 --file rpc %}.  The SOAP
+message is an XML-formatted message composed of an envelope inside which a
+header and a body are provided.  The body of the message contains the request
+and response of the message, which is transmitted over HTTP or SMTP.  The
+benefit of such a protocol is that it provides the flexibility for transmission
+over multiple transport protocol, though parsing such messages could become a
+bottleneck.
+
+#### Thrift
+
+Thrift is an RPC system created by Facebook and now part of the Apache
+Foundation {% cite thrift --file rpc %}. It is a language-agnostic IDL by which
+one generates the code for the client and server. It provides the opportunity
+for compressed serialization by customizing the protocol and the transport after
+the description file has been processed.
+
+#### Finagle
+
+Finagle was generated by Twitter and is an RPC system written in Scala and can
+run on a JVM.  It is based on three object types: Service objects, Filter
+objects and Future objects {% cite finagle --file rpc %}. The Future objects act
+by asynchronously being requested for a computation that would return a response
+at some time in the future.  The Service objects are an endpoint that will
+return a Future upon processing a request.  A Filter object transforms requests
+for further processing in case additional customization is required from a
+request.
+
+#### Open Network Computing RPC (ONC RPC)
+
+ONC was originally introduced as SunRPC {%cite sunrpc --file rpc %} for the Sun
+NFS.  The Sun NFS system had a stateless server, with client-side caching,
+unique file-handlers, and supported NFS read, write, truncate, unlink, etc
+operations. However, SunRPC was later revised as ONC in 1995 {%cite rfc1831
+--file rpc %} and then in 2009 {%cite rfc5531 --file rpc %}. The IDL used in
+ONC(and SunRPC) is External Data Representation (XDR), a serialization mechanism
+specific to networks communication and therefore, ONC is limited to applications
+like Network File Systems.
+
+#### Mobile Assistance Using Infrastructure (MAUI)
+
+The MAUI project {% cite maui --file rpc %}, developed by Microsoft is a
+computation offloading system for mobile systems. It's an automated system that
+offloads a mobile code to a dedicated infrastructure in order to increase the
+battery life of the mobile, minimize the load on the programmer and perform
+complex computations offsite. MAUI uses RPC as the communication protocol
+between the mobile and the infrastructure.
+
+#### gRPC
+
+gRPC has been built as a collaboration between Google and Square as a public
+replacement of Stubby, ARCWire, and Sake {% cite Apigee --file rpc %}.  The IDL
+for gRPC is Protocol Buffers(also referred as ProtoBuf).
```

Comment:
@heathermiller Could we just reference the  [gRPC chapter](https://github.com/heathermiller/dist-prog-book/blob/master/chapter/1/gRPC.md), since I go into detail about Protobuf there.

Let me know what you think.

Thanks,
`p